### PR TITLE
main-phase1-integration-shootToAmp-groupCommand

### PR DESCRIPTION
**Pull request by Kerenkam**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
+                new SequentialCommandGroup(
+                        new InstantCommand(() -> System.out.println("In Seq")),
+                        new ParallelDeadlineGroup(
+                                new WaitUntilCommand(()-> armSystem.reachedATargetAngle(RobotMap.ARM_AMP_ANGLE)),
+                                new ArmMoveToAmp(armSystem),
+                                new InstantCommand(() -> System.out.println("In ParDead"))
+                        ),
+                        new InstantCommand(() -> System.out.println("After ParDead")),
+                        new OuttakeCommand(intakeSystem),
+                        new InstantCommand(intakeSystem.getCurrentCommand()::cancel)
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: why is this necessary?

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
+
+        JoystickButton Ybutton = new JoystickButton(operatorController, XboxController.Button.kY.value);
+        //shooterSystem.getCurrentCommand().cancel();
+
+        ShooterPID shooterCommand = new ShooterPID(shooterSystem, 5000);
+        ArmMoveToShooterCommand armCommand = new ArmMoveToShooterCommand(armSystem);
+
+        shootToAmp = new ParallelCommandGroup(
+                new SequentialCommandGroup(
+                        new InstantCommand(() -> System.out.println("In Seq")),
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: what are these prints for?

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
         armSystem = new ArmSystem();
-        xboxController = new XboxController(0);
-
-        POVButton dPadUp = new POVButton(xboxController, 0);
-        POVButton dPadDown = new POVButton(xboxController, 180);
+        driveController = new XboxController(0);
+        operatorController = new XboxController(1);
+
+        JoystickButton Ybutton = new JoystickButton(operatorController, XboxController.Button.kY.value);
+        //shooterSystem.getCurrentCommand().cancel();
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: either uncomment or delete this line. Don't keep commented code.

File: src/main/java/frc/robot/Robot.java [hunk 4]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
+                                new WaitUntilCommand(()-> armSystem.reachedATargetAngle(RobotMap.ARM_AMP_ANGLE)),
+                                new ArmMoveToAmp(armSystem),
+                                new InstantCommand(() -> System.out.println("In ParDead"))
+                        ),
+                        new InstantCommand(() -> System.out.println("After ParDead")),
+                        new OuttakeCommand(intakeSystem),
+                        new InstantCommand(intakeSystem.getCurrentCommand()::cancel)
+                )
+        );
+        new WaitUntilCommand(()-> armSystem.reachedATargetAngle(RobotMap.ARM_AMP_ANGLE));
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: this isn't inside the command group

File: src/main/java/frc/robot/Robot.java [hunk 5]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
-
-        POVButton dPadUp = new POVButton(xboxController, 0);
-        POVButton dPadDown = new POVButton(xboxController, 180);
+        driveController = new XboxController(0);
+        operatorController = new XboxController(1);
+
+        JoystickButton Ybutton = new JoystickButton(operatorController, XboxController.Button.kY.value);
+        //shooterSystem.getCurrentCommand().cancel();
+
+        ShooterPID shooterCommand = new ShooterPID(shooterSystem, 5000);
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: this is unused

File: src/main/java/frc/robot/Robot.java [hunk 6]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
+        ArmMoveToShooterCommand armCommand = new ArmMoveToShooterCommand(armSystem);
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: this is unused

File: src/main/java/frc/robot/Robot.java [hunk 7]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
         intakeSystem = new IntakeSystem();
         armSystem = new ArmSystem();
-        xboxController = new XboxController(0);
-
-        POVButton dPadUp = new POVButton(xboxController, 0);
-        POVButton dPadDown = new POVButton(xboxController, 180);
+        driveController = new XboxController(0);
+        operatorController = new XboxController(1);
+
+        JoystickButton Ybutton = new JoystickButton(operatorController, XboxController.Button.kY.value);
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: this is unused

File: src/main/java/frc/robot/Robot.java [hunk 8]
```diff
@@ -24,32 +19,55 @@ public class Robot extends TimedRobot {
...
+        //shooterSystem.getCurrentCommand().cancel();
+
+        ShooterPID shooterCommand = new ShooterPID(shooterSystem, 5000);
+        ArmMoveToShooterCommand armCommand = new ArmMoveToShooterCommand(armSystem);
+
+        shootToAmp = new ParallelCommandGroup(
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: so basically: move arm to 130 degrees, push note out and done?

File: src/main/java/frc/robot/subsystems/ArmSystem.java [hunk 9]
```diff
@@ -26,6 +26,7 @@ public ArmSystem() {
     }
 
     public boolean reachedATargetAngle(double targetAngle) {
+        System.out.println(getAbsEncoderPositionDegrees() + " - " + MathUtil.isNear(targetAngle, getAbsEncoderPositionDegrees(), NEAR_ANGLE_TOLERANCE));
```
Comment (tomtzook) [inline src/main/java/frc/robot/subsystems/ArmSystem.java]: what's this print for?

File: src/main/java/frc/robot/subsystems/ShooterSystem.java [hunk 10]
```diff
@@ -84,6 +85,10 @@ public void rotatePID(double targetRPM){
         pid.setReference(targetRPM, CANSparkBase.ControlType.kVelocity);
     }
 
+    public boolean reachedRPM(double rpm){
```
Comment (tomtzook) [inline src/main/java/frc/robot/subsystems/ShooterSystem.java]: unused

File: src/main/java/frc/robot/Robot.java [hunk 11]
```diff
@@ -79,7 +97,28 @@ public void autonomousPeriodic() {
 
     @Override
     public void testInit() {
+        ShooterPID shooterPID = new ShooterPID(shooterSystem,5000);
```
Comment (tomtzook) [inline src/main/java/frc/robot/Robot.java]: why did you add all these commands?

**Review by tomtzook — CHANGES_REQUESTED** [review]
There's a lot of lines that I'm not sure why they exist

File: src/main/java/frc/robot/Robot.java (hunk 11 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: these commands were there before, I added mine at the end. I'll delete them

File: src/main/java/frc/robot/Robot.java (hunk 8 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: yes, isnt it?

File: src/main/java/frc/robot/Robot.java (hunk 3 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: done

File: src/main/java/frc/robot/Robot.java (hunk 2 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: to see that the command is finished, delete this?

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: to make sure this command is not running on the intake subsystem

File: src/main/java/frc/robot/Robot.java (hunk 4 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: I thought this was supposed to be outside. do i add it as the last line in the command group?

File: src/main/java/frc/robot/Robot.java (hunk 5 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: done

File: src/main/java/frc/robot/Robot.java (hunk 7 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/Robot.java]: done

File: src/main/java/frc/robot/subsystems/ShooterSystem.java (hunk 10 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/subsystems/ShooterSystem.java]: i didnt use this subsystem, should i still delete?

File: src/main/java/frc/robot/subsystems/ArmSystem.java (hunk 9 above)
Comment (Kerenkam) [reply src/main/java/frc/robot/subsystems/ArmSystem.java]: honestly idk, delete this print?

**Review by tomtzook — APPROVED** [review]
